diff --git a/src/runtime/plugin.go b/src/runtime/plugin.go
index a61dcc3b5d..eb3b87192a 100644
--- a/src/runtime/plugin.go
+++ b/src/runtime/plugin.go
@@ -48,12 +48,6 @@ func plugin_lastmoduleinit() (path string, syms map[string]any, errstr string) {
 			throw("plugin: new module data overlaps with previous moduledata")
 		}
 	}
-	for _, pkghash := range md.pkghashes {
-		if pkghash.linktimehash != *pkghash.runtimehash {
-			md.bad = true
-			return "", nil, "plugin was built with a different version of package " + pkghash.modulename
-		}
-	}
 
 	// Initialize the freshly loaded module.
 	modulesinit()
